runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -76,46 +69,28 @@ protected ILogData removeFromQueue(NavigableSet<Long> queue) {
                 return null;
             }
 
-            try {
-                // Because the discovery mechanism implemented by this class
-                // does not require to read log entries in advance (only requests
-                // the stream's full address map, without reading the actual data), entries can be read in
-                // batches whenever we have a cache miss. This allows next reads
-                // to be serviced immediately, rather than reading one entry at a time.
-                ld = read(currentRead, queue);
-
-                if (queue == getCurrentContext().readQueue && ld != null) {
-                    // Validate that the data entry belongs to this stream, otherwise, skip.
-                    // This verification protects from sequencer regression (tokens assigned in an older epoch
-                    // that were never written to, and reassigned on a newer epoch)
-                    if (ld.containsStream(this.id) && ld.getType() == DataType.DATA) {
-                        addToResolvedQueue(getCurrentContext(), currentRead, ld);
-                        readNext = false;
-                    } else {
-                        log.trace("getNextEntry[{}]: the data for address {} does not belong to this stream. Skip.",
-                                this, currentRead);
-                        // Invalid entry (does not belong to this stream). Read next.
-                        readNext = true;
-                    }
-                } else {
+            // Because the discovery mechanism implemented by this class
+            // does not require to read log entries in advance (only requests
+            // the stream's full address map, without reading the actual data), entries can be read in
+            // batches whenever we have a cache miss. This allows next reads
+            // to be serviced immediately, rather than reading one entry at a time.
+            ld = read(currentRead, queue);
+
+            if (queue == getCurrentContext().readQueue && ld != null) {
+                // Validate that the data entry belongs to this stream, otherwise, skip.
+                // This verification protects from sequencer regression (tokens assigned in an older epoch
+                // that were never written to, and reassigned on a newer epoch)
+                if (ld.containsStream(this.id) && ld.getType() == DataType.DATA) {
+                    addToResolvedQueue(getCurrentContext(), currentRead, ld);
                     readNext = false;
+                } else {
+                    log.trace("getNextEntry[{}]: the data for address {} does not belong to this stream. Skip.",
+                            this, currentRead);
+                    // Invalid entry (does not belong to this stream). Read next.
```

Comment:
We should add (does not belong to this stream or is compacted). Read next.

Reply:
I don't understand why the compacted or irrelevant addresses into the resolved queue.